Fix underscore in netcdf_fortran migrator

xylar opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * license_file entry is missing, but is required.

xylar commented:
@CJ-Wright, I fixed the underscore in the `netcdf-fortran` migrator.  I also added to the comment in the example to try to clarify.  Please let me know if this seems correct to you.

Does anything else need to happen with the migrator (like bumping the `migration_number`) or will this fix be sufficient and the bot will not be confused?

ocefpaf commented on recipe/migrations/netcdf-fortran-4.5.x.yaml:
```diff
@@ -7,5 +7,5 @@ __migrator:
     1
 
-netcdf-fortran:
+netcdf_fortran:
```
Thanks. I actually explained that in your PR #341 but totally missed it here. (It is hard to read `-` vs `_`!)
